File: week7/methodoverload.cpp
#include <iostream>
using namespace std;
class Sum{
    public:
    int add(int a, int b){
        return a+b;
    }
    double add(double a, double b){
        return a+b;
    }
    int add(int a, int b, int c){
        return a+b+c;
    }
};
int main(){
    
    Sum obj;
    cout << "add(int 17, int 9) : " << obj.add(17,9) << endl;
    cout << "add(double 42.12, double 12.32) : " << obj.add(42.12,12.32) << endl;
    cout << "add(int 40, int 9, int 14) : " << obj.add(40,9,14) << endl;
    
}

File: week7/methodoverload.java
class A {
    public void mul(int a, int b) {
        System.out.println(a + " X " + b + " = " + (a * b));
    }

    public void mul(int a, int b, int c) {
        System.out.println(a + " X " + b + " X " + c + " = " + (a * b * c));
    }

    public void mul(double a, double b) {
        System.out.println(a + " X " + b + " = " + (a * b));
    }

    public void mul(double a, double b, double c) {
        System.out.println(a + " X " + b + " X " + c + " = " + (a * b * c));
    }
}

class methodOverloading {
    public static void main(String[] args) {
        A obj = new A();
        obj.mul(17,9);
        obj.mul(42.12,32.12);
        obj.mul(2.9, 1.2, 3.6);
    }
}
